### PR TITLE
fix docker rmi stucking

### LINKED ISSUE #39739: "docker rmi -f" is stuck
**What happened:**
 my execution “docker rmi -f imageXXX” is stuck when some other container in the same host has connected to an abnormal nfs server, and i'm very sure there's no any container using this imageXXX at time.

i notice:
https://github.com/moby/moby/blob/fade624f1696a46071b812606f4ddb0c6d2e4d16/daemon/images/image_delete.go#L369-L382

here, all containers will be checked  if running and then if the containter being checked has the identical imageid to the deleting imgid .
https://github.com/moby/moby/blob/fade624f1696a46071b812606f4ddb0c6d2e4d16/container/state.go#L239-L244
and there is global lock.  That's the reason. i think maybe adjusting the statement like that  will be better:
```go
return  c.ImageID == imgID && c.IsRunning()
```

**What you expected to happen:**
no stuck when running docker rmi .
**How to reproduce it (as minimally and precisely as possible):**
1.mount some containers to nfs server.
2.make nfs server unaccessible.
3.docker rmi any container.
Environment:

version:
any version
Cloud provider or hardware configuration:
OS (e.g: cat /etc/os-release): Ubuntu 16.04 LTS 
Kernel (e.g. uname -a):4.4.0-21-generic #37-Ubuntu 
Install tools:
Network plugin and version (if this is a network-related bug):
Others:

### PR DESCRIPTION
**Pull request by yanghaichao12**
Signed-off-by: Haichao Yang yang.haichao@zte.com.cn
fixes https://github.com/moby/moby/issues/39739

- What I did
  figure out https://github.com/moby/moby/issues/39739
- How I did it
 swapping code https://github.com/moby/moby/blob/fade624f1696a46071b812606f4ddb0c6d2e4d16/daemon/images/image_delete.go#L372

- How to verify it

- Description for the changelog

Fix docker rmi stuck in case of misconfigured system (such as dead NFS share).

- A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

**Comment by yanghaichao12** [comment]
Derek add label: area/images

**Comment by yanghaichao12** [comment]
retest

**Comment by yanghaichao12** [comment]
retest

**Comment by yanghaichao12** [comment]
retest this please

**Comment by yanghaichao12** [comment]
retest this please

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ vendor/pkg/
 go-test-report.json
 profile.out
 junit-report.xml
+.idea/
```
Comment (thaJeztah) [inline .gitignore]: Can you remove this change? It is not related to the fix, and we don't add editor-specific ignores here (see https://github.com/moby/moby/blob/d4306e4b331b49f086b9bec22421879c3e58072d/.gitignore#L1-L3)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thank you; I left a comment below.

Could you also change your DCO sign-off to [use your real name](https://github.com/moby/moby/blob/0aa167d0dfd2bc1296cd3e8bfde7765f795bd561/CONTRIBUTING.md#sign-your-work), not your GitHub handle (`yanghaichao12`)? 


```
Signed-off-by: yanghaichao12 ....
```

**Comment by yanghaichao12** [comment]
@thaJeztah hi, i have change DCO and remove the content in .gitignore added by me

**Comment by thaJeztah** [comment]
Hm; perhaps something went wrong with the DCO sign-off; I still see this in the commit message 😅 ;

```
Signed-off-by: yanghaichao12 <yang.haichao@zte.com.cn>
```

**Comment by yanghaichao12** [comment]
@thaJeztah hi， how about for now？
```shell
Signed-off-by: yanghaichao <yang.haichao@zte.com.cn>
```

**Comment by thaJeztah** [comment]
It's one word, or `Yang Haichao`  ?

Oh! 🙈 Looks like your local copy was out of date, and you pulled in an extra commit in the rebase. Can you rebase once more? (make sure your fork is up-to-date, and the extra commit should disappear)

**Comment by yanghaichao12** [comment]
in fact, Yang is my last name.....,  i have modified  it

**Review by thaJeztah — APPROVED** [review]
LGTM

Thank you! sorry for all the hassle with the sign-off 🤗

**Comment by yanghaichao12** [comment]
thank you too :-)

**Comment by yanghaichao12** [comment]
retest this please

**Comment by thaJeztah** [comment]
Windows RS1 failure is unrelated

@yongtang ptal 🤗

**Review by yongtang — APPROVED** [review]
LGTM, Thanks for the fix!

**Comment by yanghaichao12** [comment]
@thaJeztah @yongtang   Thank you！

**Pull request merged**